pyanaconda/payload/install_tree_metadata.py

```diff
@@ -0,0 +1,239 @@
+
+        for retry_count in range(0, MAX_TREEINFO_DOWNLOAD_RETRIES + 1):
+            if retry_count > 0:
+                time.sleep(next(xdelay))
+            # Downloading .treeinfo
+            log.info("Trying to download '.treeinfo'")
+            (response, ret_code[0]) = self._download_treeinfo_file(session, url, ".treeinfo",
+                                                                   headers, proxies, sslverify)
+            if response:
+                break
+            # Downloading treeinfo
+            log.info("Trying to download 'treeinfo'")
+            (response, ret_code[1]) = self._download_treeinfo_file(session, url, "treeinfo",
+                                                                   headers, proxies, sslverify)
+            if response:
+                break
+
+            # The [.]treeinfo wasn't downloaded. Try it again if [.]treeinfo
+            # is on the server.
+            #
+            # Server returned HTTP 404 code -> no need to try again
+            if (ret_code[0] is not None and ret_code[0] == 404
+                    and ret_code[1] is not None and ret_code[1] == 404):
+                response = None
+                log.error("Got HTTP 404 Error when downloading [.]treeinfo files")
+                break
+            if retry_count < MAX_TREEINFO_DOWNLOAD_RETRIES:
+                # retry
+                log.info("Retrying repo info download for %s, retrying (%d/%d)",
+                         url, retry_count + 1, MAX_TREEINFO_DOWNLOAD_RETRIES)
+            else:
+                # run out of retries
+                err_msg = ("Repo info download for %s failed after %d retries" %
+                           (url, retry_count))
+                log.error(err_msg)
+                raise IOError("Can't get .treeinfo file from the url {}".format(url))
+
+        if response:
+            # get the treeinfo contents
+            text = response.text
+
+            response.close()
+
+            self._tree_info.loads(text)
+            self._path = url
+            return True
+
+        return False
+
+    @staticmethod
+    def _download_treeinfo_file(session, url, file_name, headers, proxies, verify):
+        try:
+            result = session.get("%s/%s" % (url, file_name), headers=headers,
+                                 proxies=proxies, verify=verify)
+            # Server returned HTTP 4XX or 5XX codes
+            if 400 <= result.status_code < 600:
+                log.info("Server returned %i code", result.status_code)
+                return None, result.status_code
+            log.debug("Retrieved '%s' from %s", file_name, url)
+        except requests.exceptions.RequestException as e:
+            log.info("Error downloading '%s': %s", file_name, e)
+            return (None, None)
+        return result, result.status_code
+
+    def _clear(self):
+        """Clear metadata repositories."""
+        self._meta_repos = []
+        self._path = ""
```

Comment:
Maybe `self._tree_info` should be cleared as well.

Reply:
Yep it should. I totally forgot that.